lib/cloud/clients.go

```diff
@@ -942,3 +1029,108 @@ func (c *TestCloudClients) GetAzureRunCommandClient(subscription string) (azure.
 func (c *TestCloudClients) Close() error {
 	return nil
 }
+
+// AWSSessionCacheKeyBuilder is a helper struct wrapper around a string builder
+// that is used to efficiently build cache keys for a given region and, optionally,
+// a chain of AWS assumed-role sessions.
+type AWSSessionCacheKeyBuilder struct {
+	sb        *strings.Builder
+	roleCount int
+}
+
+// NewAWSSessionCacheKeyBuilder constructs a new cache key builder intialized
+// with a specific AWS region.
+func NewAWSSessionCacheKeyBuilder(region string) AWSSessionCacheKeyBuilder {
```

Comment:
The `NewAWSSessionCacheKeyBuilder` and `AWSSessionCacheKeyBuilder` object is used only in test. Can we split the implementation from tests ?

Reply:
this was because I hadn't finished cleaning up after refactoring cloud clients. I didn't have time yesterday, but I wanted to get feedback on the refatored cloud clients overnight. I've removed cachekey builder in fc800e15f75f7ed8f38b88b6aba00a756c66d144